registry/remote/example_test.go

```diff
@@ -0,0 +1,312 @@
+	if err != nil {
+		panic(err) // Handle error
+	}
+	host = u.Host
+	http.DefaultClient = ts.Client()
+
+	os.Exit(m.Run())
+}
+
+// ExampleRepository_Tags gives example snippets for listing tags in a repository.
+func ExampleRepository_Tags() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	repo.Tags(ctx, func(tags []string) error {
+		for _, tag := range tags {
+			fmt.Println(tag)
+		}
+		return nil
+	})
+
+	// Output:
+	// tag1
+	// tag2
+}
+
+// ExampleRepository_Push gives example snippets for pushing a blob.
+func ExampleRepository_Push() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	mediaType, content := ocispec.MediaTypeImageLayer, []byte("Example blob content") // Setup input: 1) media type and 2)[]byte content
+	desc := ocispec.Descriptor{                                                       // Assemble a descriptor
+		MediaType: mediaType,                 // Set mediatype
+		Digest:    digest.FromBytes(content), // Calculate digest
+		Size:      int64(len(content)),       // Include content size
+	}
+	repo.Push(ctx, desc, bytes.NewReader(content)) // Push the blob
+
+	fmt.Println("Push finished")
+	// Output:
+	// Push finished
+}
+
+// ExampleRepository_Resolve_byTag gives example snippets for resolving a tag.
+func ExampleRepository_Resolve_byTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	descriptor, err := repo.Resolve(ctx, tag) // Resolve digest to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(descriptor.Digest)
+	fmt.Println(descriptor.Size)
+
+	// Output:
+	// sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec
+	// 20
+}
+
+// ExampleRepository_Resolve_byDigest gives example snippets for resolving a digest.
+func ExampleRepository_Resolve_byDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	descriptor, err := repo.Resolve(ctx, digest) // Resolve digest to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(descriptor.Size)
```

Comment:
Would recommend adding the mediaType since it’s typically used more than size. But showing both would be nice.